waku/v2/protocol/waku_store/protocol.nim

```diff
@@ -72,7 +72,16 @@ proc initProtocolHandler(ws: WakuStore) =
     info "received history query", peerId=conn.peerId, requestId=requestId, query=request
     waku_store_queries.inc()
 
-    let responseRes = ws.queryHandler(request)
+    var responseRes: HistoryResult
+    try:
+      responseRes = ws.queryHandler(request)
+    except Exception:
+      error "history query failed", peerId= $conn.peerId, requestId=requestId, error=getCurrentExceptionMsg()
+
+      let response = responseRes.toRPC()
+      let rpc = HistoryRPC(requestId: requestId, response: some(response))
+      await conn.writeLp(rpc.encode().buffer)
+      return
```

Comment:
LGTM!
Just a couple of doubts:

* Is this part intended to be inside the 'except' or should it be outside?

* Is 'conn' properly set at this point?

Reply:
> * Is this part intended to be inside the 'except' or should it be outside?

It is intended to be part of the except block since we respond to the node by writing to the connection.

> * Is 'conn' properly set at this point?

Yes, it should be available to write the response. The callback/handler function raises the exception here, but the connection is available.